### PR TITLE
Add some generic tools to improve reusability

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
This pull request provides some useful enhancers to speed up coding and reuse code. 
 - A generic component for toolbars (with buttons configurations, glyphs and tooltip management)
 - A tooltip enhancer. To avoid to add OverlayTrigger stuff every time.
 - A generic empty page component and enhancer (that allows to simply render a configurable empty page on some conditions
 - A Sidebar tool with good defaults, to reuse.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Feature
 - [x] Code style update (formatting, local variables)

**What is the current behavior?** (You can also link to an open issue here)
Nothing.

**What is the new behavior?**
Developers can use the new tools to implement a more coherent UI and reusing most of the code.

**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [x] No

**Other information**:
**Toolbar**
code: 
image
result: 
image

**Empty State**
code: 
image
result: 
image

### CODE REVIEW

File: web/client/components/misc/EmptyView.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * @param  {string|node} [content]             Additional content for the empty view (e.g. buttons...)
+ */
+module.exports = ({
+        opacity= 0.45,
+        glyph="info-sign",
+        title,
+        description,
+        content
+    } = {}) =>
+        (<div style={{textAlign: "center", position: "absolute", width: "100%"}} >
```
Comment (mbarto) [inline web/client/components/misc/EmptyView.jsx]: Let's use className and css

File: web/client/components/misc/EmptyView.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * @class EmptyView
+ * @memberof components.misc
+ * @param  {Number} [opacity=0.45]      The opactity
+ * @param  {String} [glyph="info-sign"] The icon glyph
+ * @param  {string|node} [title]               The title of the advice to display
+ * @param  {string|node} [description]         The description to display
+ * @param  {string|node} [content]             Additional content for the empty view (e.g. buttons...)
+ */
+module.exports = ({
+        opacity= 0.45,
```
Comment (mbarto) [inline web/client/components/misc/EmptyView.jsx]: css?

File: web/client/components/misc/enhancers/emptyState.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+const React = require('react');
+const {branch} = require('recompose');
+const DefaultEmptyComponent = require("../EmptyView");
+
+
+/**
+* Empty State enhancer. Enhances an object adding a (localizable) tooltip.
```
Comment (mbarto) [inline web/client/components/misc/enhancers/emptyState.jsx]: I think this documentation is copied from tooltip.jsx

Comment (offtherailz) [reply web/client/components/misc/enhancers/emptyState.jsx]: good catch. Going to fix it

File: web/client/components/misc/EmptyView.jsx (hunk 2 above)
Comment (offtherailz) [reply web/client/components/misc/EmptyView.jsx]: I will place it in css. Anyway I should keep the possibility to override the main div style (and then the stylesheet opacity) to allow quick fix of edge cases.

Comment (mbarto) [reply web/client/components/misc/EmptyView.jsx]: Having a style property (with empty defaults) is ok

**Pull request merged**